sql/hive/src/main/scala/org/apache/spark/sql/hive/security/HiveDelegationTokenProvider.scala

```diff
@@ -99,7 +99,12 @@ private[spark] class HiveDelegationTokenProvider
         s"$principal at $metastoreUri")
 
       doAsRealUser {
-        val hive = Hive.get(conf, classOf[HiveConf])
+        val hive = conf match {
+          case hiveConf: HiveConf =>
+            Hive.getWithoutRegisterFns(hiveConf)
```

Comment:
we will have to nunuc https://github.com/apache/spark/blob/116255d10989fd87d1eb3347bb279df207b1b5c4/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L204-L211 approach to allow other versions of hive when buidling. cc @AngersZhuuuu FYI

Reply:
Ok, I thought at first that code that has nothing to do with `HiveClientImpl` does not need to be compatible with lower versions. I also use reflection to solve compatibility problems.